### ISSUE TITLE
[HOLD for payment 2025-02-04] [$125] mWeb - 2FA - In 2FA magic code page, after refresh briefly error message is displayed

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54009]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** V9. 0.75-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Issue reported by:** Applause Internal Team
**Device used:** Redmi note 10s android 13
**App Component:** User Settings

## Action Performed:
Pre-condition: Login with unverified gmail account
1. Go to https://staging.new.expensify.com/home
2. Go to settings- security - two factor authentication
3. Enter incorrect magic code
4. Refresh the page

## Expected Result:
In 2FA magic code page, after refresh briefly error message must not be displayed.

## Actual Result:
In 2FA magic code page, after refresh briefly error message is displayed.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/2df08823-e574-42a1-809b-adaaefc48f22

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021867646284163514204</li>
        <li>Upwork Job ID: 1867646284163514204</li>
        <li>Last Price Increase: 2025-01-10</li>
<li>Automatic offers: </li>
<ul>
<li>dominictb | Reviewer | 105714118</li>
<li>nyomanjyotisa | Contributor | 105714119</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @OfstadC / @michaelkwardrop</details>

**Comment by OfstadC** [comment]
I'm currently OoO - but will be back on Tuesday - Reassigning to get the ball rolling, but happy to take back on Tuesday

**Comment by michaelkwardrop** [comment]
@dominictb please attempt reproduction and review the above proposals - thanks!

**Comment by michaelkwardrop** [comment]
@dominictb lmk if this needs to wait until the new year! 🎄

**Comment by huult** [comment]
@dominictb Could you review my proposal? thanks

**Comment by FitseTLT** [comment]
> @nyomanjyotisa's [proposal](https://github.com/Expensify/App/issues/54009#issuecomment-2539132006) looks promising. Onyx update is the async action and calling func in did-mount happens **after** rendering the component.
> 
> So using `canShowError` and clear the error on mount are the good ideas to solve the issue and be consistent with [this PR](https://github.com/Expensify/App/pull/52303)
> 
> But I just wonder:
> 
> 1. Should we call `clearValidateCodeActionError` along with `clearError` like what we did [here](https://github.com/Expensify/App/blob/9603c2f4a29ace5a158948a84f4b129710b62737/src/components/ValidateCodeActionModal/ValidateCodeForm/BaseValidateCodeForm.tsx#L183-L184)?
> 2. If we use `canShowError`, I don't think we need to clear error in clean-up function. Please correct me if I miss something.

@dominictb Why are we even considering a problem that is hypothetical raised linked to other PR. I have uploaded a video on my proposal as POC that the problem @nyomanjyotisa is claiming doesn't even exist. The approach I am suggesting is used in other places in our code base such as [here](https://github.com/Expensify/App/blob/ca4107b428ca1c48b5b0dc204683c226d42fa06f/src/pages/Search/SearchHoldReasonPage.tsx#L51-L52) [here](https://github.com/FitseTLT/App/blob/ca4107b428ca1c48b5b0dc204683c226d42fa06f/src/pages/iou/HoldReasonPage.tsx#L69-L70) @nyomanjyotisa proposal is my proposal plus a solution for a problem that doesn't even exist from my testing. Can you provide an evidence of the brief display of the error occuring in this issue's case?

**Comment by FitseTLT** [comment]
>If you can prove the issue doesn't happen anymore (by plain text not just record the video), I'll consider the decision again. Thanks 🙏

I am not saying my video should be taken as evidence but if you guys are sure the [problem](https://github.com/Expensify/App/pull/52303#issuecomment-2470033635) can also happen here give me a proof otherwise what is the point of applying a workaround solution for an edge case problem that's not even there.
The explanation that I can give you here is 
>BTW useOnyx is late in loading data I remember an issue arising on mWeb of not auto-populating tag on iou confirmation page when withFullTransactionOrNotFound HOC was migrated to use useOnyx from withOnyx and transaction was being undefined on useEffect run on render that used to work well with withOnyx and regarding the pr https://github.com/Expensify/App/pull/52303#issuecomment-2470033635 linked by @nyomanjyotisa you can see that the useOnys migration commit was made later. It might be because we are getting the error data from useOnyx that I am not reproducing the issue he is claiming and the problem wouldn't occur here because we would be clearing the error even before it is read from onyx in mWeb.

**Comment by FitseTLT** [comment]
> It doesn't really convince me. Can you show me the code?

@dominictb Just proceed with your review and let the assigned engineer decide whether we should apply a fix for an invisible bug or not 👍

**Comment by michaelkwardrop** [comment]
HNY @dominictb! Do we have a proposal we are moving forward with?

**Comment by huult** [comment]
> @huult Dismissing the error is the common technique we're using in the App so we should do the same.

@dominictb Yes, I think we should do the same. However, I believe the proposal you liked is not a valid RCA (I am not sure which API they mentioned). Could you please take a look at my proposal again? The RCA is valid and detailed, and the solution works well. Thank you.

**Comment by michaelkwardrop** [comment]
@dominictb are we ready to move forward with @nyomanjyotisa's proposal?

**Comment by michaelkwardrop** [comment]
@dominictb looks good to me! I agree with the problem/ solution statements, and if the code looks good to you I give this a 🟢

**Comment by OfstadC** [comment]
@dominictb Please complete BZ checklist prior to payment date 😃 Thanks!

**Comment by OfstadC** [comment]
## Payment Summary
- @nyomanjyotisa [paid](https://www.upwork.com/nx/wm/workroom/39530028/overview) $125
- @dominictb [paid](https://www.upwork.com/nx/wm/workroom/39658900/overview) $125
